src/sinks/azure_monitor_logs/sink.rs

```diff
@@ -0,0 +1,214 @@
+
+use crate::sinks::prelude::*;
+
+use super::service::AzureMonitorLogsRequest;
+
+pub struct AzureMonitorLogsSink<S> {
+    batch_settings: BatcherSettings,
+    encoding: JsonEncoding,
+    service: S,
+    protocol: String,
+}
+
+impl<S> AzureMonitorLogsSink<S>
+where
+    S: Service<AzureMonitorLogsRequest> + Send + 'static,
+    S::Future: Send + 'static,
+    S::Response: DriverResponse + Send + 'static,
+    S::Error: Debug + Into<crate::Error> + Send,
+{
+    pub fn new(
+        batch_settings: BatcherSettings,
+        transformer: Transformer,
+        service: S,
+        time_generated_key: Option<OwnedValuePath>,
+        protocol: String,
+    ) -> Self {
+        Self {
+            batch_settings,
+            encoding: JsonEncoding::new(transformer, time_generated_key),
+            service,
+            protocol,
+        }
+    }
+
+    async fn run_inner(self: Box<Self>, input: BoxStream<'_, Event>) -> Result<(), ()> {
+        input
+            .batched(self.batch_settings.into_byte_size_config())
+            .request_builder(
+                None,
+                AzureMonitorLogsRequestBuilder {
+                    encoding: self.encoding,
+                },
+            )
+            .filter_map(|request| async {
+                match request {
+                    Err(error) => {
+                        emit!(SinkRequestBuildError { error });
+                        None
+                    }
+                    Ok(req) => Some(req),
+                }
+            })
+            .into_driver(self.service)
+            .protocol(self.protocol.clone())
+            .run()
+            .await
+    }
+}
+
+#[async_trait::async_trait]
+impl<S> StreamSink<Event> for AzureMonitorLogsSink<S>
+where
+    S: Service<AzureMonitorLogsRequest> + Send + 'static,
+    S::Future: Send + 'static,
+    S::Response: DriverResponse + Send + 'static,
+    S::Error: Debug + Into<crate::Error> + Send,
+{
+    async fn run(
+        self: Box<Self>,
+        input: futures_util::stream::BoxStream<'_, Event>,
+    ) -> Result<(), ()> {
+        self.run_inner(input).await
+    }
+}
+
+/// Customized encoding specific to the Azure Monitor Logs sink, as the API does not support full
+/// 9-digit nanosecond precision timestamps.
+#[derive(Clone, Debug)]
+struct JsonEncoding {
+    time_generated_key: Option<OwnedValuePath>,
+    encoder: (Transformer, Encoder<Framer>),
+}
+
+impl JsonEncoding {
+    pub fn new(transformer: Transformer, time_generated_key: Option<OwnedValuePath>) -> Self {
+        Self {
+            time_generated_key,
+            encoder: (
+                transformer,
+                Encoder::<Framer>::new(
+                    CharacterDelimitedEncoder::new(b',').into(),
+                    JsonSerializerConfig::default().build().into(),
+                ),
+            ),
+        }
+    }
+}
+
+impl crate::sinks::util::encoding::Encoder<Vec<Event>> for JsonEncoding {
```

Comment:
There's too many `Encoder`s in the project. We should sort this out at some point..